Sources/TypeChecker/TypeChecker+Expressions.swift

```diff
@@ -130,6 +132,42 @@ extension TypeChecker {
                                       expression: argument.expression))
         }
       }
+    } else if case .matchedFunction(let matchingFunction) =
+      environment.matchFunctionCall(functionCall,
+                                    enclosingType: functionCall.identifier.enclosingType ?? enclosingType,
+                                    typeStates: typeStates,
+                                    callerProtections: callerProtections,
+                                    scopeContext: passContext.scopeContext!) {
+
+      if let externalCall = passContext.externalCall {
+        // check value parameter (type)
+        if matchingFunction.declaration.isPayable {
+          if let valueParameter: FunctionArgument = externalCall.getHyperParameter(parameterName: "value") {
+            let parameterType = environment.type(of: valueParameter.expression,
+              enclosingType: enclosingType,
+              typeStates: typeStates,
+              callerProtections: callerProtections,
+              scopeContext: passContext.scopeContext!)
+
+            if parameterType != .userDefinedType("Wei") {
+              diagnostics.append(.valueParameterWithWrongType(valueParameter))
+            }
+          }
+        }
+
+        // check gas parameter (type)
+        if let gasParameter: FunctionArgument = externalCall.getHyperParameter(parameterName: "gas") {
+          let parameterType = environment.type(of: gasParameter.expression,
+            enclosingType: enclosingType,
+            typeStates: typeStates,
+            callerProtections: callerProtections,
+            scopeContext: passContext.scopeContext!)
+
+          if parameterType != .basicType(.int) {
```

Comment:
Should gas no also be specified in Wei? Gas is measured in Wei.

Reply:
No, resulting issue is #107